modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/IgniteDataStorageMetricsSelfTest.java

```diff
@@ -389,6 +392,124 @@ public void testWalWrittenBytes() throws Exception {
+    }
+
+    /**
+     * Check whether WAL is reporting correct usage when archiving is performed.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testWalTotalSizeWithArchive() throws Exception {
+        IgniteEx n = startGrid(0, (Consumer<IgniteConfiguration>)cfg ->
+            cfg.getDataStorageConfiguration().setWalSegmentSize((int)(2 * U.MB)));
+
+        n.cluster().state(ACTIVE);
+        awaitPartitionMapExchange();
+
+        for (int i = 0; i < 10; i++)
+            n.cache("cache").put(ThreadLocalRandom.current().nextLong(), new byte[(int)(32 * U.KB)]);
+
+        while (walMgr(n).lastArchivedSegment() < 3)
+            n.cache("cache").put(ThreadLocalRandom.current().nextLong(), new byte[(int)(32 * U.KB)]);
+
+        WALDisableContext walDisableCtx = n.context().cache().context().walState().walDisableContext();
+        assertNotNull(walDisableCtx);
+
+        setFieldValue(walDisableCtx, "disableWal", true);
+
+        assertTrue(walDisableCtx.check());
+        assertNull(walMgr(n).log(new DataRecord(emptyList())));
+
+        FileWriteAheadLogManager walMgr = walMgr(n);
+
+        SegmentRouter router = walMgr.getSegmentRouter();
+
+        Thread.sleep(1000); //wait to avoid race condition with FileWriteAheadLogManager.FileArchiver#allocateRemainingFile where wal.tmp files are created.
+        long totalSize = walMgr.totalSize(FileWriteAheadLogManager.loadFileDescriptors(router.getWalWorkDir()));
+
+        assertTrue(router.hasArchive());
+
+        totalSize += walMgr.totalSize(FileWriteAheadLogManager.loadFileDescriptors(router.getWalArchiveDir()));
+
+        assertTrue(totalSize == dbMgr(n).persistentStoreMetrics().getWalTotalSize());
+        assertTrue(totalSize == dsMetricsMXBean(n).getWalTotalSize());
+        assertTrue(totalSize == ((LongGauge)dsMetricRegistry(n).findMetric("WalTotalSize")).value());
+    }
+
+    /**
+     * Check whether Wal is reporting correct usage when WAL Archive is turned off.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testWalTotalSizeWithArchiveTurnedOff() throws Exception {
+        IgniteEx n = startGrid(0, (Consumer<IgniteConfiguration>)cfg ->
+            cfg.getDataStorageConfiguration().setWalArchivePath(cfg.getDataStorageConfiguration().getWalPath()).setWalSegmentSize((int)(2 * U.MB)));
+
+        n.cluster().state(ACTIVE);
+        awaitPartitionMapExchange();
+
+        for (int i = 0; i < 10; i++)
+            n.cache("cache").put(ThreadLocalRandom.current().nextLong(), new byte[(int)(32 * U.KB)]);
+
+        WALDisableContext walDisableCtx = n.context().cache().context().walState().walDisableContext();
+        assertNotNull(walDisableCtx);
+
+        setFieldValue(walDisableCtx, "disableWal", true);
+
+        assertTrue(walDisableCtx.check());
+        assertNull(walMgr(n).log(new DataRecord(emptyList())));
+
+        assertEquals(-1, walMgr(n).lastArchivedSegment());
```

Comment:
I think that this is not necessary, you can do this:
`assertFalse(router.hasArchive());`

Reply:
agree